### DISCUSSION TITLE
orientation data flipped upon export?

### DISCUSSION THREAD

**Discussion by gabriele-casale** [mtex-toolbox/mtex#1476] [Ask Anything] [answered]
When I rotate orientation data and then export to a new .ctf file, for some reason the orientations, but not the spatial data are flipped about the vertical axis. The maps plot identically, but the pole figures are flipped. Anyone have any ideas?

Before Export
![raw](https://user-images.githubusercontent.com/108618427/177042170-65c07cab-ca57-472d-8bc1-17caf9a9d001.jpg)

After Export
![rot](https://user-images.githubusercontent.com/108618427/177042176-58f26758-519e-4dde-bc73-88c2a8111e32.jpg)

**Comment by zmichels** [comment]
Are you rotating the whole EBSD dataset (as opposed to a variable containing orientations)

Can you give a code example of the rotation you applied?

Zach

**Reply by gabriele-casale** [reply]
Hey Zach! 
I am only rotating the orientation data, but I am applying the rotation to the whole dataset using 'keepXY'. That seems to work fine, and both the odfs and maps plot how I would expect. Its only when I export, reload, and calculate new odfs that the orientation data is flipped. The images I included in the original post are the exact same dataset post rotation. The second image is after I reload the data. 

Here is my rotation: 
ebsd = rotate(ebsd, rotation.byAxisAngle(xvector,Xrot*degree), 'keepXY');
ebsd = rotate(ebsd, rotation.byAxisAngle(yvector,Yrot*degree), 'keepXY');

**Comment by kilir** [comment, marked as answer]
Hi,
if you export data in mtex to a ctf, the "inconsistent" coordinate system which it might have had originally ( = the reason one might use the presets e.g. 'convertEuler2SpatialReferenceFrame' or your own rotation) is not undone. Accordingly, if you import your new ctf no correction should be needed.

Cheers,
Rüdiger

**Reply by gabriele-casale** [reply]
Thanks Rudiger!
This is exactly the problem. I was converting the exported dataset using the now unnecessary correction that you mentioned ('convertEuler2SpatialReferenceFrame'). Problem solved.
